src/3d/qgspointcloudlayer3drenderer.h

```diff
@@ -27,9 +27,73 @@
 
 class QgsPointCloudLayer;
 #include "qgspointcloud3dsymbol.h"
+#include "qgsfeature3dhandler_p.h"
 
 #ifndef SIP_RUN
 
+/**
+ * \ingroup core
+ * \class QgsPointCloud3DRenderContext
+ *
+ * Encapsulates the render context for a 3D point cloud rendering operation.
+ *
+ * \since QGIS 3.18
+ */
+class CORE_EXPORT QgsPointCloud3DRenderContext : public Qgs3DRenderContext
```

Comment:
CORE_EXPORT ? That looks wrong... and I think it should not even be _3D_EXPORT - this should be a private implementation class not exposed to the public API

Reply:
Sorry I forgot it there when I copied from the QgsPointCloud2DRenderContext